src/pages/History.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect, useContext } from "react";
+import NormalNavbar from "../components/NormalNavbar";
+import { VideoContext } from "../context/Data";
+import { Link } from "react-router-dom";
+
+const History = () => {
+  const token =
+    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiIxMjVkM2MzNy01MjcwLTQ5NjgtODQ0MC1iZTM3ZDFhNTE5OGUiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.DPS9hLIaykSx9V9SwXsOhWgWQ7nk8MtTyumcWlbYamM";
+  const { videoState, dispatch } = useContext(VideoContext);
+  const getHistoryData = async () => {
+    const getData = await fetch("/api/user/history", {
+      method: "GET",
+      headers: {
+        authorization: token,
+      },
+    });
+    const convertedJSON = await getData.json();
+    console.log(convertedJSON);
+    dispatch({
+      type: "historyVideos",
+      payload: { value: convertedJSON.history },
+    });
+  };
+  const deleteHistoryVideo = async (id) => {
+    try {
+      const deleteData = await fetch(`/api/user/history/${id}`, {
+        method: "DELETE",
```

Comment:
These history service functions can be moved in a seperate file. Just a suggestion.